staging/src/k8s.io/client-go/rest/client.go

```diff
@@ -110,6 +111,16 @@ type RESTClient struct {
 // NewRESTClient creates a new RESTClient. This client performs generic REST functions
 // such as Get, Put, Post, and Delete on specified paths.
 func NewRESTClient(baseURL *url.URL, versionedAPIPath string, config ClientContentConfig, rateLimiter flowcontrol.RateLimiter, client *http.Client) (*RESTClient, error) {
+	return NewRESTClientWithLogger(klog.Background(), baseURL, versionedAPIPath, config, rateLimiter, client)
+}
+
+// NewRESTClientWithLogger creates a new RESTClient. This client performs generic REST functions
+// such as Get, Put, Post, and Delete on specified paths.
+//
+// The methods of the RESTClient support contextual logging, i.e. they will look for a logger
+// in the context passed to them. The logger here is only used during construction of the
+// client and its configuration.
+func NewRESTClientWithLogger(logger klog.Logger, baseURL *url.URL, versionedAPIPath string, config ClientContentConfig, rateLimiter flowcontrol.RateLimiter, client *http.Client) (*RESTClient, error) {
```

Comment:
There is no need to switch to this *in Kubernetes* because Kubernetes doesn't use the feature implementation which logs, therefore the `NewRESTClient` function doesn't get the usual `Contextual logging: use NewRESTClientWithLogger` comment.

The alternative is to use `klog.Background` in `features.go` - or remove those log calls altogether. I am not sure how useful they are.

Thoughts?